crates/net/network/src/manager.rs

```diff
@@ -542,15 +543,22 @@ impl<N: NetworkPrimitives> NetworkManager<N> {
     }
 
     /// Handles a received Message from the peer's session.
+    #[allow(clippy::needless_pass_by_ref_mut)]
     fn on_peer_message(&mut self, peer_id: PeerId, msg: PeerMessage<N>) {
         match msg {
             PeerMessage::NewBlockHashes(hashes) => {
+                #[cfg(feature = "scroll")]
+                debug!(target: "net", ?hashes, "incoming NewBlockHashes peer message");
+                #[cfg(not(feature = "scroll"))]
                 self.within_pow_or_disconnect(peer_id, |this| {
                     // update peer's state, to track what blocks this peer has seen
                     this.swarm.state_mut().on_new_block_hashes(peer_id, hashes.0)
                 })
```

Comment:
Is the `scroll` feature flag suitable here? The `scroll` feature flag has implications for the state commitment logic.